connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/entities/ConnectorPluginInfo.java

```diff
@@ -79,11 +82,21 @@ public int hashCode() {
 
     @Override
     public String toString() {
-        final StringBuilder sb = new StringBuilder("ConnectorPluginInfo{");
-        sb.append("className='").append(className).append('\'');
-        sb.append(", type=").append(type);
-        sb.append(", version='").append(version).append('\'');
-        sb.append('}');
-        return sb.toString();
+        return "ConnectorPluginInfo{" + "className='" + className + '\'' +
+                ", type=" + type.toString() +
+                ", version='" + version + '\'' +
+                '}';
+    }
+
+    public static final class NoVersionFilter {
+        @Override
+        public boolean equals(Object obj) {
+            return DelegatingClassLoader.UNDEFINED_VERSION.equals(obj);
+        }
+
+        @Override
+        public int hashCode() {
+            return super.hashCode();
+        }
```

Comment:
Do we need this method declaration at all? Looks like it might have been left in accidentally while trying to wrestle with SpotBugs?
```suggestion
```

Reply:
Without this method, the compilation fails. To be honest, I'm not sure if there's a way to avoid it.

Reply:
Ah yeah, compiler warning about overriding `equals` but not `hashCode`. Think we should leave a comment about that just so others don't wonder the same thing?

Reply:
Yes that's a good idea